sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -720,6 +722,14 @@ class Analyzer(
     }
   }
 
+  case class ResolveNamespace(catalogManager: CatalogManager)
+    extends Rule[LogicalPlan] with LookupCatalog {
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case UnresolvedNamespace(CatalogAndNamespace(catalog, ns)) =>
+        ResolvedNamespace(catalog.asNamespaceCatalog, ns)
```

Comment:
I have a question. We have the following in `ResolveSessionCatalog.scala`:
```scala
case ShowTablesStatement(Some(CatalogAndNamespace(catalog, ns)), pattern)
  if isSessionCatalog(catalog) =>
```
Do we plan to add these statements here and not resolve `UnresolvedNamespace` and let `ResolveSessionCatalog` handle them?